src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4816,6 +4857,24 @@ private ImmutableArray<Symbol> CreateReducedExtensionMethodIfPossible(BoundDeleg
         /// <param name="node">The node.</param>
         public abstract AwaitExpressionInfo GetAwaitExpressionInfo(AwaitExpressionSyntax node);
 
+        /// <summary>
+        /// If the given token is within a preprocessing directive, gets the preprocessing symbol info for it.
+        /// Define and undefine directive trivia parents are supported too.
+        /// </summary>
+        /// <param name="token">Preprocessing symbol syntax token.</param>
+        public new PreprocessingSymbolInfo GetPreprocessingSymbolInfo(SyntaxToken token)
+        {
+            var parent = token.Parent as CSharpSyntaxNode;
+            CheckSyntaxNode(parent);
+
+            if (parent.Kind() is SyntaxKind.DefineDirectiveTrivia or SyntaxKind.UndefDirectiveTrivia)
+                return CreatePreprocessingSymbolInfo(token);
+
+            if (parent is IdentifierNameSyntax identifier)
+                return GetPreprocessingSymbolInfo(identifier);
+
+            return PreprocessingSymbolInfo.None;
+        }
```

Comment:
this does not feel like a needed api.  the caller could do thsi work themselves.